### ISSUE TITLE
Can't connect to signet

### ISSUE THREAD

**Issue by cd2357** [sparrowwallet/sparrow#384]
Trying to start Sparrow v1.5.5 using signet with

```
/Applications/Sparrow.app/Contents/runtime/Contents/Home/bin/sparrow --network signet
```

The welcome wizard pops up. Clicking through successfuly, until the step "Connecting to a private Electrum Server".

If at that step I click "Configure server" I get :

```
2022-01-04 17:33:58,227 ERROR Exception in thread "JavaFX Application Thread"
java.lang.IllegalArgumentException: bound must be positive
	at java.base/java.util.Random.nextInt(Unknown Source)
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.preferences.ServerPreferencesController.initializeView(Unknown Source)
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.preferences.PreferencesController.setPreferencePane(Unknown Source)
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.preferences.PreferencesController.lambda$initializeView$0(Unknown Source)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(Unknown Source)
	at javafx.base/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.beans.property.ReadOnlyObjectPropertyBase.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.beans.property.ReadOnlyObjectWrapper.fireValueChangedEvent(Unknown Source)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.markInvalid(Unknown Source)
	at javafx.base/javafx.beans.property.ObjectPropertyBase.set(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleGroup$3.set(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleGroup$3.set(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleGroup.selectToggle(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleButton$1.invalidated(Unknown Source)
	at javafx.base/javafx.beans.property.BooleanPropertyBase.markInvalid(Unknown Source)
	at javafx.base/javafx.beans.property.BooleanPropertyBase.set(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleButton.setSelected(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleGroup.setSelected(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleGroup$3.set(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleGroup$3.set(Unknown Source)
	at javafx.controls/javafx.scene.control.ToggleGroup.selectToggle(Unknown Source)
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.preferences.PreferencesController.lambda$selectGroup$1(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.mac.MacApplication._enterNestedEventLoopImpl(Native Method)
	at javafx.graphics/com.sun.glass.ui.mac.MacApplication._enterNestedEventLoop(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.Application.enterNestedEventLoop(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.EventLoop.enter(Unknown Source)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(Unknown Source)
	at javafx.graphics/javafx.stage.Stage.showAndWait(Unknown Source)
	at javafx.controls/javafx.scene.control.HeavyweightDialog.showAndWait(Unknown Source)
	at javafx.controls/javafx.scene.control.Dialog.showAndWait(Unknown Source)
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.MainApp.start(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
```

and the UI is not initialized.

Also tried it with

```
/Applications/Sparrow.app/Contents/runtime/Contents/Home/bin/sparrow -n signet
```

with the same result.

Mainnet and testnet work fine, only getting this for signet.

**Comment by cd2357** [comment]
Got this error for all versions down to v1.5.2.

On v1.5.2 it correctly tells me I need to setup the connection to the local Bitcoin Core. After I navigate in the UI and do that (Preferences > Server) then it works on v1.5.2. In higher versions navigating to this screen shows a blank UI window and the stacktrace above.

**Comment by craigraw** [comment]
Thanks, fixed in f1b3f7d.
